docs/pages/7_0_0_release_notes.md

```diff
@@ -123,50 +180,69 @@ We are shipping the following rules:
+  an introduction to new features in XPath 3.1.
 
-#### Node stream API
+#### JavaScript support
 
-This version includes a powerful API to navigate trees, similar in usage to the Java 8 Stream API:
-```java
-node.descendants(ASTMethodCall.class)
-    .filter(m -> "toString".equals(m.getMethodName()))
-    .map(m -> m.getQualifier())
-    .filter(q -> TypeTestUtil.isA(String.class, q))
-    .foreach(System.out::println);
-```
+The JS specific parser options have been removed. The parser now always retains comments and uses version ES6.
+The language module registers a couple of different versions. The latest version, which supports ES6 and also some
+new constructs (see [Rhino](https://github.com/mozilla/rhino)]), is the default. This should be fine for most
+use cases.
 
-A pipeline like shown here traverses the tree lazily, which is more efficient than traversing eagerly to put all descendants in a list. It is also much easier to change than the old imperative way.
+#### Language versions
 
-To make this API as accessible as possible, the {% jdoc core::lang.ast.Node %} interface has been fitted with new methods producing node streams. Those methods replace previous tree traversal methods like `Node#findDescendantsOfType`. In all cases, they should be more efficient and more convenient.
+We revisited the versions that were defined by each language module. Now many more versions are defined for each
+language. In general, you can expect that PMD can parse all these different versions. There might be situations
+where this fails and this can be considered a bug. Usually the latest version is selected as the default
+language version.
 
-See {% jdoc core::lang.ast.NodeStream %} for more details.
+The language versions can be used to mark rules to be useful only for a specific language version via
+the `minimumLanguageVersion` and `maximumLanguageVersion` attributes. While this feature is currently only used by
+the Java module, listing all possible versions enables other languages as well to use this feature.
 
+Related issue: [[core] Explicitly name all language versions (#4120)](https://github.com/pmd/pmd/issues/4120)
 
-#### JavaScript support
+#### API
 
-The JS specific parser options have been removed. The parser now always retains comments and uses version ES6.
-The language module registers a couple of different versions. The latest version, which supports ES6 and also some
-new constructs (see [Rhino](https://github.com/mozilla/rhino)]), is the default. This should be fine for most
-use cases.
+The API of PMD has been growing over the years and needed some cleanup. The goal is, to
+have a clear separation between a well-defined API and the implementation, which is internal.
+This should help us in future development.
+
+#### Compatibility and migration notes
+
+PMD 7 requires Java 8 or above to execute.
+
+TODO - what to look out for. Hints and known problems.
```

Comment:
Maybe we can postpone this section, we have #4294 planned...
Or maybe keep it very short

* CLI changed: Custom scripts need to be updated
* Java module revamped: Custom rules need to be updated
* Deleted rules: Custom rulesets need to be reviewed. See below for a list of new and deleted rules.
* XPath 1.0 support is removed, `violationSuppressXPath` now requires XPath 2.0 or 3.1: Custom rulesets need to be reviewed.
* Custom rules using rulechains: some API changes (around `RuleTargetSelector`, `AbstractRule#buildTargetSelector()`)